File: src/nodes/var_decl.zig
const nd = @import("node.zig");
const bytecode = @import("../common/bytecodes.zig");

pub const NodeVarDecl = struct {
    name: []const u8,
    vtype: bytecode.ApicaTypeBytecode,
    expression: *nd.Node,

    pub fn init(name: []const u8, vtype: bytecode.ApicaTypeBytecode, expr: *nd.Node) NodeVarDecl {
        return NodeVarDecl{ .name = name, .vtype = vtype, .expression = expr };
    }

    pub fn get_name(self: *const NodeVarDecl) []const u8 {
        return self.name;
    }

    pub fn get_vtype(self: *const NodeVarDecl) bytecode.ApicaTypeBytecode {
        return self.vtype;
    }

    pub fn get_expression(self: *const NodeVarDecl) *nd.Node {
        return self.expression;
    }
};

File: src/nodes/var_decl.rs
use apica_common::bytecodes::ApicaTypeBytecode;
use crate::nodes::node::Node;

pub struct NodeVarDecl {
    name: String,
    value_kind: ApicaTypeBytecode,
    expression: Node,
}

impl NodeVarDecl {
    pub fn init(name: String, value_kind: ApicaTypeBytecode, expression: Node) -> NodeVarDecl {
        NodeVarDecl { name, value_kind, expression }
    }
    
    pub fn get_name(&self) -> &String {
        &self.name
    }
    
    pub fn get_value_kind(&self) -> &ApicaTypeBytecode {
        &self.value_kind
    }
    
    pub fn get_expression(&self) -> &Node {
        &self.expression
    }
}